UI/window-basic-properties.cpp

```diff
@@ -117,12 +117,20 @@ OBSBasicProperties::OBSBasicProperties(QWidget *parent, OBSSource source_)
 
 	enum obs_source_type type = obs_source_get_type(source);
 	uint32_t caps = obs_source_get_output_flags(source);
-	bool drawable_type = type == OBS_SOURCE_TYPE_INPUT ||
+	bool live_drawable_type = type == OBS_SOURCE_TYPE_INPUT ||
 		type == OBS_SOURCE_TYPE_SCENE;
-
-	if (drawable_type && (caps & OBS_SOURCE_VIDEO) != 0)
-		connect(preview.data(), &OBSQTDisplay::DisplayCreated,
+	bool drawablePreview = (caps & OBS_SOURCE_VIDEO) != 0;
+	if (drawablePreview) {
```

Comment:
Because the results of both "fail" branches are the same, I think this `if` statement can be rewritten like this:
```
	if (drawablePreview && live_drawable_type) {
		preview->show();
		connect(preview.data(), &OBSQTDisplay::DisplayCreated,
				addDrawCallback);
	} else {
		preview->hide();
	}
```

Reply:
Those were definitely by QT Designer, output_flags are not changable, they should be constant b/c they're set as part of the plugins...though maybe there's some plugin out there that changes those flags?

As for drawable_type vs live_drawable_type there's an opportunity for transitions to have a "preview", but with dummy sources likely w/ prerecorded or generated video, I prefixed the current variable w/ "live" to potentially distinguish that difference if it ends up being added in the future.

It's also why I wrote the if block in that way, distinguishing the top logic "can this thing even draw to a preview" from "this thing has something to preview, and here's the callback to use".